### DISCUSSION TITLE
Package.json - single pre hook for multiple scripts

### DISCUSSION THREAD

**Discussion by yairvol** [npm/rfcs#560] [RRFC]
Hey, I'd like to define a pre-hook (or post) once in the package.json, and apply it to several scripts.
The issue is you can't - you must duplicate the hooks, which seems redundant (I have 7!).
For example:
![image](https://user-images.githubusercontent.com/87264348/159902809-66d548e9-4b57-4821-bafc-d8608a7275ad.png)

Will be replaced with:
![image](https://user-images.githubusercontent.com/87264348/159903510-0f948a04-f2c5-48a3-8aca-bef4a2ef7eb4.png)

**Comment by darcyclarke** [comment]
This is a nice idea - there's other improvements we need to make to script definitions & execution as well. If you're willing to write an RFC for this we could discuss it a bit more in our weekly calls.

**Reply by yairvol** [reply]
Happy to hear, how do I write one though?
